src/pages/index.js

```diff
@@ -5,86 +5,235 @@ import Section from "../components/Section.js";
+  defaultText = "Save"
+) {
+  popupInstance.setLoading(true, loadingText);
+
+  return request()
+    .then(() => {
+      popupInstance.close();
+    })
+    .catch((error) => {
+      console.error("Error:", error);
+    })
+    .finally(() => {
+      popupInstance.setLoading(false, defaultText);
+    });
+}
+
+/* DELETE Place Function */
+function handleDeleteClick(cardID, cardElement) {
+  deletePlaceModal.setSubmitHandler(() => {
+    function makeRequest() {
+      return api.removePlace(cardID).then(() => {
+        cardElement.remove();
+      });
+    }
+
+    handleSubmit(makeRequest, deletePlaceModal, "Removing...", "Yes");
+  });
+
+  deletePlaceModal.open();
+}
+
+/* Create Cards */
+function getCardElement(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeReact
+  );
+  return card.getView();
+}
+
+/* GET Cards */
+let section;
+api
+  .getInitialCards()
+  .then((items) => {
+    section = new Section(
+      {
+        items: items,
+        renderer: (cardData) => {
+          const cardElement = getCardElement(cardData);
+          return cardElement;
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+/* GET Profile */
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+api
+  .getUserInfo()
+  .then((profileData) => {
+    userInfo.setUserInfo(profileData);
+  })
+  .catch((err) => console.error(err));
+
+/* PATCH Profile Edit Function */
+function handleProfileEditSubmit(profileInputValues) {
+  function makeRequest() {
+    return api
+      .editUserInfo(profileInputValues.title, profileInputValues.description)
+      .then((updatedUserData) => {
+        userInfo.setUserInfo(updatedUserData);
+      });
+  }
+
+  handleSubmit(makeRequest, editProfileModal);
+}
+
+/* PATCH Profile Avatar Function */
+function handleAvatarSubmit({ url }) {
+  function makeRequest() {
+    return api.updateAvatar(url).then(() => {
+      userInfo.updateAvatar(url);
+    });
+  }
+
+  handleSubmit(makeRequest, updateAvatarModal, "Saving...").then(() => {
+    updateAvatarModal.reset();
+    updateAvatarModal.setLoading(true, "Save");
+  });
+}
+
+/* POST Add Place Function */
+function handleNewPlaceSubmit(placeCardData) {
+  function makeRequest() {
+    const cardData = {
+      name: placeCardData.title,
+      link: placeCardData.url,
+    };
+    return api
+      .addNewPlace(cardData.name, cardData.link)
+      .then((newPlaceCard) => {
+        const cardElement = getCardElement({
+          name: newPlaceCard.name,
+          link: newPlaceCard.link,
+          _id: newPlaceCard._id,
+        });
+        section.addItem(cardElement);
+      });
+  }
+
+  handleSubmit(makeRequest, addPlaceModal, "Saving...").then(() => {
+    addPlaceModal.reset();
+    addPlaceModal.setLoading(true, "Save");
```

Comment:
You shouldn't use `then` after calling `handleSubmit` because it contains everything inside itself (so you can't continue the code after `catch` and `finally` which is inside `handleSubmit`

You need to reset in `then` and `setLoading` is inside  `handleSubmit` in `finally` already